File: python/62_Unique_Paths.py
class Solution:
    # @param {integer} m
    # @param {integer} n
    # @return {integer}
    def uniquePaths(self, m, n):
        if not m or not n:
            return 0
        if m==1 or n==1:
            return 1
        below=[1]*n
        cur=[1]*n
        for r in xrange(m-2,-1,-1):
            for c in xrange(n-1,-1,-1):
                if c==n-1:
                    cur[c]=1
                else:
                    cur[c]=below[c]+cur[c+1]
            below=cur
        return cur[0]

File: Java/62_Unique_Paths.java
public class Solution {
    public int uniquePaths(int m, int n) {
        int[] pre = new int[n];
        int[] cur = new int[n];
        int i,j;
        for(i=0;i<n;i++){
            pre[i]=1;
        }
        for(i=1;i<m;i++){
            for(j=0;j<n;j++){
                cur[j]=pre[j];
                if(j>0){
                    cur[j]+=cur[j-1];
                }
            }
            for(j=0;j<n;j++){
                pre[j]=cur[j];
                cur[j]=0;
            }
        }
        return pre[n-1];
    }
}